scripts/commands/checkLinks.ts

```diff
@@ -0,0 +1,140 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { globby } from "globby";
+import { existsSync, readFileSync } from 'fs';
+import path from 'node:path';
+import markdownLinkExtractor from 'markdown-link-extractor';
+
+const DOCS_ROOT = "./docs"
+const CONTENT_FILE_EXTENSIONS = [".md", ".mdx", ".ipynb"]
+
+class Link {
```

Comment:
It would be super valuable to have some unit tests for `Link.resolve()` and `Link.check()`. They don't have any side effects anymore and they're now pure functions, so it should be easy to test. 

To do so, you'd add a file `scripts/commands/linkChecker.ts` or something like that. Copy this code over. Then import it from your commands file. And add a test file similar to `dedupeIds.test.ts`.

`npm test` to ensure the test works.